docs/contributing/testing.md

```diff
@@ -0,0 +1,112 @@
+- [run it from the command line](https://standardjs.com/#usage)
+
+Make sure you fix any errors that the linters report.
+
+## 2. Check your changes in our ‘review app’
+
+In your project folder, run `npm start`, then go to `localhost:3000` in your browser to open our 'review app'.
+
+Check that:
+
+- the examples in the review app look right and behave correctly with your changes
+- your design is consistent with the rest of GOV.UK Frontend
+
+## 3. Test in supported browsers and assistive technology
+
+You should test that your contribution works:
+
+- in [recommended browsers](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in)
+- with [recommended assistive technologies](https://www.gov.uk/service-manual/technology/testing-with-assistive-technologies#what-to-test)
+- in [Internet Explorer 8](/docs/installation/supporting-internet-explorer-8.md), 9 and 10 - components do not need to look perfect
+- when your users [override colours in Windows, Firefox and Chrome](https://accessibility.blog.gov.uk/2018/08/01/supporting-users-who-change-colours-on-gov-uk/)
+
+## 4. Run our automated tests
+
+In your project folder, run `npm start` to run our automated tests.
+
+If a test fails, you should check your code for any errors, then fix any tests you need to.
+
+## 5. Write new tests
+
+You should write new tests if you’ve done any of the following:
+
+- created a new component
+- changed or added to a component’s JavaScript code
+- changed or added to a component’s Nunjucks macro
+- created or updated a Sass mixin or function
+
+If you’re not sure how to write new tests, ask a developer to help you or [contact us](https://design-system.service.gov.uk/#support).
+
+### If you created a component
+
+Create the following files in the `src/govuk/components` folder:
+
+- `<COMPONENT>/<COMPONENT>.test.js` - to test functionality
+- `<COMPONENT>/template.test.js` - to test the Nunjucks macro
+
+Where `<COMPONENT>` is the name of the component you created.
+
+You can use the existing files in the `src/govuk/components` folder as templates for your new files.
+
+### If you changed or added to a component
+
+In the `src/govuk/components` folder, update or add tests to:
+
+- `<COMPONENT>/<COMPONENT>.test.js` - if you updated functionality
+- `<COMPONENT>/template.test.js` - if you updated the Nunjucks macro
+
+Where `<COMPONENT>` is the name of the component you changed or added to.
+
+### If you created or updated a Sass mixin or function
+
+Update or add tests In `src/govuk/<ELEMENT>/grid.test.js`, where `<ELEMENT>` is the [element you’re creating or updating a mixin for](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/README.md).
+
+## 6. Update our snapshot tests
+
+If your component uses another component, one of our 'snapshot tests' may fail. Snapshot tests compare a component's current markup with a previously stored version.
```

Comment:
Should we link to Jest's snapshot testing docs since this is perhaps something people are not used to?